README.md

```diff
@@ -1,147 +1,181 @@
+
+## Usage
+### Model Directory Structure
 ```
 example-model/
-├── promote.sh
-└── main.R (our deployment script)
+├── deploy.R
+└── promote.sh (optional)
 ```
 
-`promote.sh` - this file is executed before your model is built. It can be used to install low-level system packages such as Linux packages
-
-`main.R` - our primary model deployment script
-
-## Building a model:
-
-Before beginning building a model, be sure to import the `promote` package:
+- [`deploy.R`](#deployr): our primary model deployment script
 
-`libary(promote)`
-
-### The `model.predict()` function
+- [`promote.sh`](#promotesh): this file is executed before your model is built. It can be used to install low-level system packages such as Linux packages
+<hr>
 
-The `model.predict` function is used o define the API endpoint for a model and is executed each time a model is called. **This is the core of the API endpoint**
+### `deploy.R`
+#### Steps
+- [Initial Setup](#setup)
+- [`model.predict`](#modelpredict)
+- [Test Data](#testing)
+- [`promote.library`](#promotelibrary)
+- [`promote.metadata`](#promotemetadata)
+- [`promote.config`](#promoteconfig)
+- [`promote.deploy`](#promotedeploy)
+<hr>
 
+#### <a name="setup"></a>Initial Setup
+Load the `promote` library that was previously installed
 ```r
-# import the promote package and define our model function
 library(promote)
-
-model.predict <- function(request) {
-  me <- request$name
-  greeting <- paste0("Hello", me, "!")
-  greeting
-}
-
-# add metadata to attach to this model version
-promote.metadata("NAME1",value1)
-promote.metadata("NAME2",value2)
-
-# specify the username, apikey and url, and then deploy
-promote.config  <- c(
-  username = "YOUR_USERNAME",
-  apikey = "YOUR_API_KEY",
-  env = "PROMOTE_URL"
-)
-
-promote.deploy("HelloWorld", confirm = FALSE)
 ```
 
-<hr>
-
-### Setting the Auth
-
-To deploy models, you'll need to add your username, API key, and URL to the `promote.config` variable
+Import your saved model object
 ```r
-promote.config <- c(
-  username = [USERNAME],
-  apikey = [APIKEY],
-  env = [URL]
-)
+# Previously saved model 'save(my_model, file = "my_model.rda")'
+load("my_model.rda")
 ```
 <hr>
 
-### `Promote`
+#### `model.predict()`
+The `model.predict` function is used to define the API endpoint for a model and is executed each time a model is called. **This is the core of the API endpoint**
 
-The `Promote` packages has 3 methods:
-- [`promote.deploy`](#promotedeploy)
-- [`promote.metadata`](#promotemetadata)
-- [`promote.library`](#promotelibrary)
+**Arguments**
+- `data`(_Data.Frame_): the data frame generated from the json sent to the deployed model
 
-### `promote.deploy()`
-
-#### Deploy a model to Alteryx Promote
+**Example**
+```r
+model.predict <- function(data) {
+  # generate predictions from the model based on the incoming dataframe
+  predict(my_model, data)
+}
+```
+<hr>
```

Comment:
Would probably remove this  <hr> on line 73 so test data is visually a part of the model.predict section, arguably might want to switch the Test Data header to simply bold. I think it's a good aside while we're talking about model.predict, but as separated visually as it is now I read it as an odd break between two method sections.